test/data_frame_test.exs

```diff
@@ -2,7 +2,76 @@ defmodule DataFrameTest do
   use ExUnit.Case
   doctest DataFrame
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  alias DataFrame.Frame
+
+  def empty_frame do
+    %Frame{values: [[]], index: [], columns: []}
+  end
+
+  def single_entry_frame do
+    DataFrame.new([[0, 0]], [:A])
+  end
+
+  describe "new/2" do
+    test "empty input is an empty Frame" do
+      # should the index be [[]]? Shouldn't it be []?
+      assert DataFrame.new([[]], []) == 
+        %Frame{values: [[]], index: [[]], columns: []}
+    end
+
+    test "a regular dataframe" do
+      # the index seems a bit counter-intuitive. Shouldn't this just be [0, 1]?
+      assert DataFrame.new([[0, 0], [1, 1]], [:A]) ==
+        %Frame{values: [[0], [1]], index: [[0], [1]], columns: [:A]}
+    end
```

Comment:
This test and the one above it revealed something a bit counter-intuitive about how the DataFrame Frame struct works. The index seems to have an extra level of nesting i.e. `[[]]` instead of `[]` and `[[0], [1]]` instead of `[0, 1]` like I would expect (since the index is always single dimensional). The tests above pass as the code exists now but I wanted to leave the comments in the PR until it is merged to gauge your thoughts on this.